parser/src/main/java/com/microsoft/gctoolkit/parser/GCLogParser.java

```diff
@@ -165,20 +163,26 @@ MemoryPoolSummary extractPermOrMetaspaceRecord(String line) {
         MemoryPoolSummary metaDataPool = null;
         if ((trace = PERM_SPACE_RECORD.parse(line)) != null) {
             String type = trace.getGroup(1).trim();
-            if ("CMS Perm".equals(type))
-                metaDataPool = extractPermGenRecord(trace);
-            else if ("PS Perm".equals(type))
-                metaDataPool = extractPermGenRecord(trace);
-            else if ("PSPermGen".equals(type))
-                metaDataPool = extractPermGenRecord(trace);
-            else if ("Perm".equals(type))
-                metaDataPool = extractPermGenRecord(trace);
-            else if ("Metaspace".equals(type)) {
-                if (trace.getGroup(2) != null) {
-                    metaDataPool = new MetaspaceRecord(trace.toKBytes(2), trace.toKBytes(4), trace.toKBytes(6));
-                } else {
-                    metaDataPool = new MetaspaceRecord(trace.toKBytes(4), trace.toKBytes(4), trace.toKBytes(6));
-                }
+            switch (type) {
```

Comment:
Add comment: These strings are pulled out of hte file and therefore isn't appropriate to become an enum

Also, collapse the case statement